cmake/VcpkgToolchain.cmake

```diff
@@ -69,7 +67,42 @@ if(_HOST_IS_WINDOWS AND NOT DEFINED VCPKG_TARGET_TRIPLET)
     endif()
 endif()
 
-# Default to using the same triplet for the host and target to prevent unnesecary
+if(_HOST_IS_DARWIN AND NOT DEFINED VCPKG_TARGET_TRIPLET)
+    if(CMAKE_OSX_ARCHITECTURES STREQUAL "x86_64")
+        set(VCPKG_TARGET_TRIPLET "x64-macos-plasma" CACHE STRING "")
+    elseif(CMAKE_OSX_ARCHITECTURES STREQUAL "arm64")
+        set(VCPKG_TARGET_TRIPLET "arm64-macos-plasma" CACHE STRING "")
+    elseif(DEFINED CMAKE_OSX_ARCHITECTURES)
+        set(VCPKG_TARGET_TRIPLET "universal-macos-plasma" CACHE STRING "")
+    endif()
+
+    # CMAKE_HOST_SYSTEM_PROCESSOR is undefined at this point
+    execute_process(
+        COMMAND uname -m
+        OUTPUT_VARIABLE HOST_SYSTEM_PROCESSOR
```

Comment:
This should probably be `_HOST_SYSTEM_PROCESSOR` (leading underscore) to match the general CMake convention of underscore prefix being a kind of "local" variable.